zoom_rooms/indico_zoom_rooms/plugin.py

```diff
@@ -0,0 +1,270 @@
+
+
+def _signal_zoom_meeting_cloned(
+    old_assoc: VCRoomEventAssociation,
+    new_assoc: VCRoomEventAssociation,
+    vc_room: VCRoom,
+    link_object: SessionBlock | Contribution | Event,
+):
+    # on clone, clone also the zoom room calendar entry
+    if new_assoc.link_object.room:
+        _handle_link_object_created(new_assoc)
+
+
+def _signal_zoom_meeting_association_attached(
+    assoc: VCRoomEventAssociation,
+    vc_room: VCRoom,
+    event: Event,
+    data: dict,
+    old_link: VCRoomEventAssociation | None,
+    new_room: bool = False,
+):
+    # ignore associations to new rooms, since that's already handled by `new_vc_room`
+    if not new_room:
+        # this is a new association to an existing room, hence a new slot has to be created
+        _handle_link_object_created(assoc)
+
+
+def _signal_zoom_meeting_association_detached(
+    assoc: VCRoomEventAssociation,
+    vc_room: VCRoom,
+    old_link: Event | Contribution | SessionBlock,
+    event: Event,
+    data: dict | None = None,
+):
+    if vc_room.type == 'zoom':
+        # when a room is detached from an event/contribution/session block, delete the corresponding zoom room entry
+        _delete_zoom_association(old_link, vc_room)
+
+
+def _signal_zoom_meeting_data_updated(vc_room: VCRoom, data: dict):
+    if vc_room.type == 'zoom' and (name := data.get('name')) and name != vc_room.name:
+        for assoc in vc_room.events:
+            if zr_id := get_zoom_room_id(assoc.link_object):
+                ZoomRoomsQueueEntry.record(ZoomRoomsAction.update, zr_id, assoc=assoc, args=OperationArgs(title=name))
+
+
+class ZoomRoomsPlugin(IndicoPlugin):
+    """Zoom Rooms
+
+    Zoom Rooms / Exchange synchronization plugin for Indico."""
+
+    configurable = True
+    settings_form = PluginSettingsForm
+    default_settings = {'debug': False, 'service_url': None, 'docs_url': '', 'token': None, 'timeout': 3}
+
+    def init(self):
+        super().init()
+        # Here we plug the various object signals into the action tracking code
+        # Consult the README.md for a summary of the logic behind
+        self.connect(signals.event.contribution_updated, _signal_link_object_updated)
+        self.connect(signals.event.session_block_updated, _signal_link_object_updated)
+        self.connect(signals.event.updated, _signal_link_object_updated)
+        self.connect(signals.event.times_changed, _signal_tt_entry_updated, sender=Contribution)
+        self.connect(signals.event.times_changed, _signal_tt_entry_updated, sender=SessionBlock)
+        self.connect(signals.vc.new_vc_room, _signal_zoom_meeting_created)
+        self.connect(signals.vc.cloned_vc_room, _signal_zoom_meeting_cloned)
+        self.connect(signals.vc.attached_vc_room, _signal_zoom_meeting_association_attached)
+        self.connect(signals.vc.detached_vc_room, _signal_zoom_meeting_association_detached)
+        self.connect(signals.vc.updated_vc_room_data, _signal_zoom_meeting_data_updated)
```

Comment:
Usually we put those signal handlers as methods in the plugin class.

Reply:
Yeah, I know. But is it really useful in this case?

Reply:
Doesn't really make a difference I guess, so mostly consistency (and being able to use `self` isntead of `ZoomRoomsPlugin.instance`, but you're not using that in any of the signal handlers anyway).

Reply:
Maybe it's my inner Rustacean showing, but I tend to dislike methods which do not use `self` at all. And class methods in Python are very messy...

Reply:
> And class methods in Python are very messy...

Agreed, I never liked certain linters (looking at you, PyCharm) suggesting people to make something a staticmethod just because it's not using self. IMHO something should be a classmethod/staticmethod when it's *meant to be called* in such a way, and not just because it's technically *possible* to call it w/o an instance...

Reply:
Hm, what about moving the functions into a separate files (handlers.py) like we did for livesync? Keeps the plugin file cleaner as well!

Reply:
Great idea.